spec/requests/api/vms_spec.rb

```diff
@@ -177,6 +177,19 @@ def update_raw_power_state(state, *vms)
       expect_single_action_result(:success => true, :message => "starting", :href => :vm_url, :task => true)
     end
 
+    it "starting a vm queues it properly" do
+      api_basic_authorize action_identifier(:vms, :start)
+      update_raw_power_state("poweredOff", vm)
+
+      run_post(vm_url, gen_request(:start))
+
+      expect_single_action_result(:success => true, :message => "starting", :href => :vm_url, :task => true)
+      expect(MiqQueue.where(:class_name  => vm.class.name,
+                            :instance_id => vm.id,
+                            :method_name => "start",
+                            :zone        => zone.name).count).to eq(1)
+    end
+
```

Comment:
unless we're testing the my_zone method, I don't think it's necessary, without the fix, this test fails (zone is not added in the queue). here the zone obtained is from the vm/instance my_zone (which is the my_zone of the vm's ems), and in the test case, the api_zone of the ems in the api  test env.